core/vm/evm.go

```diff
@@ -450,6 +450,15 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		return nil, common.Address{}, gas, ErrNonceUintOverflow
 	}
 	evm.StateDB.SetNonce(caller.Address(), nonce+1)
+
+	// Charge the contract creation init gas in verkle mode
+	if evm.chainRules.IsEIP4762 {
+		statelessGas := evm.Accesses.TouchAndChargeContractCreateInit(address.Bytes(), value.Sign() != 0)
+		if statelessGas > gas {
+			return nil, common.Address{}, gas, ErrOutOfGas
+		}
+		gas = gas - statelessGas
+	}
```

Comment:
I think that there is a slight problem with this: when we implement `FILL_COST`, we will have to check the state location - and therefore add an entry to the witness BEFORE we check for the collision. This means that we already check the collision twice.

Reply:
But isn't the FILL_COST going to be charged when the actual contract is created?
At this point, we're adding to the witness to give enough data to do the check, so a prior no FILL_COST charging?

Reply:
this problem seems independent of the fill cost. as you said, we write both the leaves no matter what.
also even if the values transfered is zero, we write to both the leaves. L474 - we create the account and set nonce to 1.

Reply:
indeed, it's also true of write - but only if the account does not exist already. Let's discuss it on the VIC on Monday.